demisto_sdk/commands/content_graph/strict_objects/base_strict_model.py

```diff
@@ -0,0 +1,138 @@
+from typing import Optional, Any, Union, Type
+
+import pydantic
+from pydantic.fields import FieldInfo
+
+from demisto_sdk.commands.common.StrEnum import StrEnum
+from pydantic import BaseModel, Extra, Field
+from demisto_sdk.commands.common.constants import Auto, MarketplaceVersions, TYPE_PWSH, TYPE_PYTHON, TYPE_JS
+
+marketplace_suffixes = [marketplace.value for marketplace in MarketplaceVersions]
+
+
+def create_dynamic_model(field_name: str, type_: Type, default: Any = ..., suffixes: list[str] = marketplace_suffixes,
+                         alias: Optional[str] = None):
+    """
+    This function creates a sub-model for avoiding duplicate lines of parsing arguments with different suffix.
+    (we have fields that are almost identical, except for the suffix.
+     for example: description:xsoar, description:marketplacev2, description:xpanse etc.)
+    Then the model inherit it for adding those fields to the root.
+
+    This is a better way than declaring on those fields manually in the root object, in this way:
+    description_xsoar: Optional[str] = Field(None, alias="description:xsoar")
+    description_marketplace_v2: Optional[str] = Field(None, alias="description:marketplacev2")
+    """
+    return pydantic.create_model(
+        f'Dynamic{field_name.title()}Model',
+        **{f'{field_name}_{suffix}': (type_, FieldInfo(default, alias=f'{alias or field_name}:{suffix}'))
+           for suffix in suffixes}, __base__=BaseStrictModel
+    )
+
+
+class BaseStrictModel(BaseModel):
+    class Config:
+        """
+        This is the definition of not allowing extra fields except those defined by the schema.
+        """
+        extra = Extra.forbid
+
+
+class CommonFields(BaseStrictModel):
+    id_: str = Field(..., alias="id")
+    version: int
+    id_xsoar: str = Field(None, alias="id:xsoar")
+    id_marketplacev2: str = Field(None, alias="id:marketplacev2")
+    id_xsoar_saas: str = Field(None, alias="id:xsoar_saas")
+    id_xsoar_on_prem: str = Field(None, alias="id:xsoar_on_prem")
+
+
+class Argument(BaseStrictModel):
+    name: str
+    required: Optional[bool] = None
+    default: Optional[bool] = None
+    description: str
+    auto: Optional[Auto] = None
+    predefined: Optional[list[str]] = None
+    is_array: Optional[bool] = Field(None, alias="isArray")
+    default_value: Optional[Any] = Field(None, alias="defaultValue")
+    secret: Optional[bool] = None
+    deprecated: Optional[bool] = None
+    type: Optional[str] = None
+    hidden: Optional[bool] = None
+    name_xsoar: Optional[str] = Field(None, alias="name:xsoar")
+    name_marketplacev2: Optional[str] = Field(None, alias="name:marketplacev2")
+    name_xpanse: Optional[str] = Field(None, alias="name:xpanse")
+    name_xsoar_saas: Optional[str] = Field(None, alias="name:xsoar_saas")
+    name_xsoar_on_prem: Optional[str] = Field(None, alias="name:xsoar_on_prem")
```

Comment:
consider moving the `name` dynamic field to this file, then importing and inheriting it in all content models, to reduce code duplications